doc/api/brotli.md

```diff
@@ -0,0 +1,404 @@
+Close the underlying handle.
+
+### brotli.flush(callback)
+
+Flush pending data. Don't call this frivolously, premature flushes negatively
+impact the effectiveness of the compression algorithm.
+
+Calling this only flushes data from the internal `brotli` state, and does not
+perform flushing of any kind on the streams level. Rather, it behaves like a
+normal call to `.write()`, i.e. it will be queued up behind other pending
+writes and will only produce output when data is being read from the stream.
+
+### brotli.setParameter(param, value)
+
+- `param` {number} A valid parameter from [Constants][]
+- `value` {number}
+
+Dynamically set a parameter.
+
+### brotli.reset()
+
+Reset the compressor/decompressor to factory defaults. Only applicable to
+the inflate and deflate algorithms.
+
+## Class: brotli.Compress
+
+Compress a stream.
+
+### new brotli.Compress(options)
+
+Creates and returns a new [Compress][] object with the given [`options`][].
+
+### Class Property: brotli.Compress.codes
+
+Error codes for compression operations.
+
+## Class: brotli.Decompress
+
+Decompress a stream.
+
+### new brotli.Decompress(options)
+
+Creates and returns a new [Decompress][] object with the given [`options`][].
+
+### Class Property: brotli.Decompress.codes
+
+Error codes for decompression operations.
+
+## brotli.constants
+
+Provides an object enumerating Brotli-related constants.
+
+## Convenience Methods
+
+<!--type=misc-->
+
+All of these take a [`Buffer`][], [`TypedArray`][], [`DataView`][],
+[`ArrayBuffer`][] or string as the first argument, an optional second argument
+to supply options to the `brotli` classes and will call the supplied callback
+with `callback(error, result)`.
+
+Every method has a `*Sync` counterpart, which accept the same arguments, but
+without a callback.
+
+### brotli.compress(buffer[, options][, callback])
+
+- `buffer` {Buffer|TypedArray|DataView|ArrayBuffer|string}
+
+Decompress a chunk of data with [Decompress][].
+If callback is omitted a Promise will be returned.
+
+### brotli.compressSync(buffer[, options])
+
+- `buffer` {Buffer|TypedArray|DataView|ArrayBuffer|string}
+
+Compress a chunk of data with [Compress][].
+
+### brotli.decompress(buffer[, options][, callback])
+
+- `buffer` {Buffer|TypedArray|DataView|ArrayBuffer|string}
+
+Decompress a chunk of data with [Decompress][].
+If callback is omitted a Promise will be returned.
+
+### brotli.decompressSync(buffer[, options])
+
+- `buffer` {Buffer|TypedArray|DataView|ArrayBuffer|string}
+
+Decompress a chunk of data with [Decompress][].
+
+[`.flush()`]: #brotli_brotli_flush_kind_callback
+[`Accept-Encoding`]: https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.3
+[`ArrayBuffer`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer
+[`brotli/encode.h`]: https://github.com/google/brotli/blob/v1.0.4/c/include/brotli/encode.h
+[`brotli/decode.h`]: https://github.com/google/brotli/blob/v1.0.4/c/include/brotli/decode.h
```

Comment:
We usually sort bottom references in ASCII order, so this should co before the `` [`brotli/encode.h`]: ``.

Reply:
References below also need some resorting.

Reply:
Don't know how I managed to mess up sorting 😆